src/docs/sphinx/dev_guide/release.rst

```diff
@@ -103,72 +102,105 @@ Typical changes that are made in a release candidate branch include:
-develop branches disagree, and we want to preserve the history. After
-merging, the release candidate branch can be deleted.
+Merge the release candidate branch into the main branch after it has been
+approved and all CI checks have passed. Do not "squash merge" as it will make
+the histories of main and develop branches disagree, and we want to preserve
+the history. After merging, the release candidate branch can be deleted.
 
-
-4: Draft a GitHub Release
+5: Draft a GitHub Release
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
 `Draft a new Release on GitHub <https://github.com/LLNL/axom/releases/new>`_
 
-#. Enter the desired tag version, e.g., v0.5.0
+#. Enter a Release title. We typically use titles of the following form *Axom-v0.3.1*
 
 #. Select **main** as the target branch to tag a release.
 
-#. Enter a Release title. We typically use titles of the following form *Axom-v0.3.1*
+#. Enter the release tag name, such as v0.5.0, and specify to create the tag 
+   when the release is published.
 
 #. Copy and paste the information for the release from the
-   ``axom/RELEASE-NOTES.md`` into the release description (omit any sections if empty).
+   ``axom/RELEASE-NOTES.md`` into the release description 
+   (omit any sections that are empty).
+
+#. Add a statement at the top of the release note, such as:
+   Please download the `Axom-*.tar.gz` tarball below, which includes all of the
+   submodules needed to build Axom. The `AxomData-*.tar.gz` tarball contains
+   extra files to put in the `data` directory only if you want to run certain
+   Axom tests.
 
 #. Publish the release. This will create a tag at the tip of the main
    branch and add corresponding entry in the
    `Releases section <https://github.com/LLNL/axom/releases>`_
 
 .. note::
 
-   GitHub will add a corresponding tarball and zip archives consisting of the
+   GitHub will add tarfile and zip archives consisting of the
    source files for each release. However, these files do not include any
-   submodules. Consequently, a tarball that includes all of the submodules is
+   submodules. Consequently, a tarfile that includes all of the submodules is
    generated manually in a separate step.
 
-5: Make a Release Tarball
+6: Make Release Tarfiles
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-* Checkout the main branch locally and run ``axom/scripts/make_release_tarball.sh --with-data``
-  Run this script from the top level ``axom`` subdirectory. This will
-  generate a two tarballs of the form ``Axom-v0.3.1.tar.gz`` and ``AxomData-v0.3.1.tar.gz``
-  consisting of the axom source and data respectively.
-
-* Upload the tarballs for the corresponding release, by going to the
-  `GitHub Releases section <https://github.com/LLNL/axom/releases>`_ and ``Edit``
-  the release created earlier.
-
-* Attach the tarball to the release.
+* After the release is published on GitHub, checkout the main branch locally
+  and run the command ``axom/scripts/make_release_tarball.sh --with-data``
```

Comment:
The script path shouldn't contain "axom/", since we're running from the axom directory.

Reply:
Good catch. Fixed.